services/workflows-service/src/data-analytics/data-analytics.service.ts

```diff
@@ -471,43 +472,57 @@ export class DataAnalyticsService {
     ) / 59);
       `,
     );
+  }
+
+  async evaluateMultipleMerchantsOneCounterparty({
+    projectId,
+    timeUnit,
+    timeAmount,
+    minimumCount,
+    excludedCounterparty = {
+      counterpartyBeneficiaryIds: [],
+      counterpartyOriginatorIds: [],
+    },
+  }: TMultipleMerchantsOneCounterparty) {
+    if (!projectId) {
+      throw new Error('projectId is required');
+    }
+
+    const conditions: Prisma.Sql[] = [
```

Comment:
This `conditions` variable makes the query much hard to read, and it's just the beginning of this query, it will evolve and become even harder to read.
I suggest writing these conditions inside the query (line 513).

Reply:
@alonp99 what do you think